platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -438,19 +386,32 @@ public void initialize(
     roleFilter = (String) options.get(ROLE_FILTER);
     roleNameAttribute = (String) options.get(ROLE_NAME_ATTRIBUTE);
     roleSearchSubtree = Boolean.parseBoolean((String) options.get(ROLE_SEARCH_SUBTREE));
-    startTls = Boolean.parseBoolean(String.valueOf(options.get(SSL_STARTTLS)));
     setBindMethod((String) options.get(BIND_METHOD));
     realm = (String) options.get(REALM);
     kdcAddress = (String) options.get(KDC_ADDRESS);
+    String connectionPoolId = (String) options.get(CONNECTION_POOL_ID);
+    installLdapConnectionPool(connectionPoolId);
+  }
 
-    if (ldapConnectionFactory != null) {
-      ldapConnectionFactory.close();
-    }
-
-    try {
-      ldapConnectionFactory = createLdapConnectionFactory(connectionURL, startTls);
-    } catch (LdapException e) {
-      LOGGER.info("Unable to create LDAP Connection Factory. LDAP log in will not be possible.", e);
+  private void installLdapConnectionPool(String connectionPoolId) {
+    BundleContext bundleContext = getContext();
+    if (bundleContext != null) {
+      try {
+        Collection<ServiceReference<GenericObjectPool>> serviceReferences =
+            bundleContext.getServiceReferences(
+                GenericObjectPool.class, String.format("(id=%s)", connectionPoolId));
+        ServiceReference<GenericObjectPool> serviceReference =
+            serviceReferences
+                .stream()
+                .findFirst()
+                .orElseThrow(
+                    () ->
+                        new IllegalStateException(
+                            "No LDAPConnectionPool service found with id:" + connectionPoolId));
+        ldapConnectionPool = bundleContext.getService(serviceReference);
+      } catch (InvalidSyntaxException | IllegalStateException e) {
+        LOGGER.info("Unable to get LDAP Connection pool. LDAP log in will not be possible.", e);
```

Comment:
✏️ this seems like a pretty severe issue.  Shouldn't this be logged at warn or even error?